apps/wallet-api-tools/src/Editor.tsx

```diff
@@ -126,52 +210,73 @@ export function Editor() {
 
         const parsedRequest = schemaRequest.parse(JSON.parse(input));
 
-        const request = {
+        const request: Request = {
           id: requestId,
           jsonrpc: "2.0",
           ...parsedRequest,
         };
 
-        const message = JSON.stringify(request, null, 2);
-        pushMessage("out", message);
+        pushOutgoingMessage(request);
 
-        transport.send(message);
+        transport.send(JSON.stringify(request, null, 2));
       } catch (error) {
         setHistory((prev) => [
-          ...prev,
           {
+            id: uuidv4(),
             date: new Date(),
             type: "error",
             value: String(error),
           },
+          ...prev,
         ]);
       }
     },
-    [setHistory, pushMessage]
+    [pushOutgoingMessage, setHistory],
   );
 
   return (
-    <>
-      <CodeMirror
-        ref={codeMirrorRef}
+    <div className="flex flex-col h-full">
+      <div
+        ref={panelRef}
+        onScroll={handleScroll}
+        className="flex-1 flex flex-col-reverse
+       overflow-y-auto p-6 pb-6 gap-y-6"
```

Comment:
We might need the [prettier-plugin-tailwindcss](https://tailwindcss.com/blog/automatic-class-sorting-with-prettier) in the wallet-api-tools app just like it's done in `examples/client-nextjs/.prettierrc`